go-fuzz/internal/writerset/writerset.go

```diff
@@ -0,0 +1,98 @@
+// Package writerset implements a mechanism to add and remove writers from a construct
+// similar to io.MultiWriter. Vendored from github.com/stephens2424/writerset
+package writerset
+
+import (
+	"io"
+	"net/http"
+	"sync"
+)
+
+// WriterSet wraps multiple writers like io.MultiWriter, but such that individual
+// writers are easy to add or remove as necessary.
+type WriterSet struct {
+	m  map[io.Writer]chan error
+	mu *sync.Mutex
+}
+
+// New initializes a new empty writer set.
+func New() *WriterSet {
+	return &WriterSet{
+		m:  make(map[io.Writer]chan error),
+		mu: &sync.Mutex{},
+	}
+}
+
+// Add ensures w is in the set.
+func (ws *WriterSet) Add(w io.Writer) <-chan error {
+	ws.mu.Lock()
+	defer ws.mu.Unlock()
+
+	c, ok := ws.m[w]
+	if ok {
+		return c
+	}
+
+	c = make(chan error, 1)
+	ws.m[w] = c
+	return c
+}
+
+// Contains determines if w is in the set.
+func (ws *WriterSet) Contains(w io.Writer) bool {
+	ws.mu.Lock()
+	defer ws.mu.Unlock()
+
+	_, ok := ws.m[w]
+	return ok
+}
+
+// Remove ensures w is not in the set.
+func (ws *WriterSet) Remove(w io.Writer) {
+	ws.mu.Lock()
+	defer ws.mu.Unlock()
+	delete(ws.m, w)
+}
+
+// Write writes data to each underlying writer. If an error occurs on an underlying writer,
+// that writer is removed from the set. The error will be sent on the channel created when
+// the writer was added.
+func (ws *WriterSet) Write(b []byte) (int, error) {
+	var err error
```

Comment:
this err is never assigned, in the loop you create a new err variable
so just return nil always